core/common/src/main/scala/org/typelevel/otel4s/context/vault/VaultContext.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+package context
+package vault
+
+import cats.Functor
+import cats.effect.Unique
+import cats.syntax.functor._
+import org.typelevel.vault.{Key => VaultKey}
+import org.typelevel.vault.Vault
+
+/** A context implementation backed by a [[org.typelevel.vault.Vault `Vault`]].
+  */
+final class VaultContext private (vault: Vault) {
```

Comment:
we do not (except in tests). I wrote it both as simply an additional implementation, to help establish that `Context` has a good API, and for potential future use in an eventual scala-only backend. I can move it to the tests to avoid confusion